### PR TITLE
SERVICES-1008 In registration service add Async Calls

### PR DESCRIPTION
**Pull request by sqreek**
Async calls to:
- SendConfirmation email
- Update AntiSpoof
- Set user preference

Also refactored WikiaHelperGateway

ping @Wikia/services-team

### CODE REVIEW

File: service/user-registration/src/main/java/com/wikia/services/userregistration/gateway/WikiaHelperGateway.java [hunk 1]
```diff
@@ -32,127 +36,107 @@
...
-      if (payload.containsKey(ALLOW_KEY)) {
-        return payload.get(ALLOW_KEY);
-      }
-    } catch (IOException e) {
-      LOGGER.error(REQUEST_FAILED_MSG, e);
-      throw new GatewayException(e.getMessage());
+      return requestPost(METHOD_CHECK_SPOOF, name).allow;
+    } catch (GatewayException e) {
+      LOGGER.error(Markers.append("username", name), "Error while checking antispoof", e);
+      return false;
```
Comment (sqreek) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/gateway/WikiaHelperGateway.java]: Is this should be a true when Gateway has some problems?

File: service/user-registration/src/main/java/com/wikia/services/userregistration/gateway/WikiaHelperGateway.java [hunk 2]
```diff
@@ -32,127 +36,107 @@
...
-    );
+  @Data
+  public static class WikiaHelperResponse {
+    @JsonProperty
+    private Boolean allow;
+    @JsonProperty
+    private String message;
+    @JsonProperty
+    private Boolean success;
   }
```
Comment (drsnyder) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/gateway/WikiaHelperGateway.java]: I know it's trivial, but I think this should be in a separate file.

File: service/user-registration/src/main/java/com/wikia/services/userregistration/service/RegistrationService.java [hunk 3]
```diff
@@ -93,13 +95,20 @@ public String registerUser(RegistrationData registrationData) throws ValidationE
...
   }
 
+  private void postRegistrationCommands(String userId,
+                                        String username,
+                                        RegistrationData registrationData) {
+
+    executorService.submit(() -> setUserPreferences(userId, registrationData));
+    executorService.submit(() -> sendConfirmationEmail(username, registrationData.getLangCode()));
+    executorService.submit(() -> updateAntiSpoof(username));
+  }
```
Comment (drsnyder) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/service/RegistrationService.java]: What happens if one or more of these throws an exception?

File: service/user-registration/src/main/java/com/wikia/services/userregistration/gateway/WikiaHelperGateway.java (hunk 2 above)
Comment (sqreek) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/gateway/WikiaHelperGateway.java]: It's like 50/50 points for keep it here or in separate file.
If none will make vote on it, I will put it in separate file

File: service/user-registration/src/main/java/com/wikia/services/userregistration/service/RegistrationService.java (hunk 3 above)
Comment (sqreek) [inline service/user-registration/src/main/java/com/wikia/services/userregistration/service/RegistrationService.java]: Exactly nothing :)
So this is a little tricky.
Expected exceptions are handled inside methods, so we don't need to worry about them.

**Comment by ArturKlajnerok** [comment]
Lgtm :+1:

**Pull request merged**